### ISSUE TITLE
Default generated certificate is no longer valid for Chrome 75+

### ISSUE THREAD

**Issue by polarathene** [containous/traefik#5006]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

Tried to verify HTTPS support was working with Traefik by using the default certificate generation before considering to generate with LetsEncrypt.

### What did you expect to see?

A webpage warning me about the certificate with the option to continue at my own risk. So that I could validate I had everything setup right.

I later tried Firefox which gave me the expected result:

From Firefox warning page:

> Websites prove their identity via certificates. Firefox does not trust this site because it uses a certificate that is not valid for test.localhost. The certificate is only valid for d94f3495596eb88e450c9038c5c00574.bcdadf74aff3e54a6503f1a8b98ba018.traefik.default.
> 
>  Error code: MOZILLA_PKIX_ERROR_SELF_SIGNED_CERT"

### What did you see instead?

An error page which gave the impression I had done something wrong.

Chrome 75 page output:

> This site can’t be reached
> 
> The webpage at https://test.localhost/ might be temporarily down or it may have moved permanently to a new web address.
> 
> ERR_SSL_KEY_USAGE_INCOMPATIBLE

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.0-alpha6
Codename:     faisselle
Go version:   go1.12.6
Built:        2019-06-18T18:25:38Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

**static config:**
```toml
[entryPoints]
  [entryPoints.web]
    address = ":80"
  [entryPoints.secure]
    address = ":443"
  [entryPoints.api]
    address = ":8080"
    
[providers]
  [providers.docker]
    endpoint = "unix:///var/run/docker.sock"
  [providers.file]
    watch = true
    filename = "/config/dynamic/traefik_dynamic.toml"
```

**dynamic config:**
```toml
[providers.file]

[http]
  [http.routers]
    [http.routers.router0]
      entryPoints = ["web"]
      service = "docker.whoami_composeProjectName" # This service is required, even though it's never used by the router
      rule = "Host(`test.localhost`)"
      middlewares=["TLS_Redirect"]

    [http.routers.router1]
      entryPoints = ["secure"]
      service = "docker.whoami_composeProjectName"
      rule = "Host(`test.localhost`)"
      [http.routers.router1.tls]

  [http.middlewares]
    [http.middlewares.TLS_Redirect.redirectscheme]
      scheme = "https"
```

**docker-compose.yml:**
```yml
version: '3'

services:
  reverse-proxy:
    image: traefik:v2.0
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - $PWD/config/traefik_static.toml:/etc/traefik/traefik.toml
      - $PWD/config/dynamic/:/config/dynamic/
    networks:
      mynetwork:
        aliases:
          - proxy.docker.localhost
  whoami:
    image: containous/whoami
    container_name: whoami
    labels:
      - "traefik.http.routers.whoami.rule=Host(`whoami.localhost`)"
    networks:
      mynetwork:
        aliases:
          - whoami.docker.localhost

networks:
  mynetwork:
    driver: bridge
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

This error only shows in logs when accessing the HTTPS route via Chrome 75, no problem with Firefox which serves the warning page.

```
reverse-proxy_1  | time="2019-06-24T06:12:14Z" level=debug msg="Serving default certificate for request: \"test.localhost\""
reverse-proxy_1  | 2019/06/24 06:12:14 server.go:3012: http: TLS handshake error from 172.18.0.1:54128: remote error: tls: illegal parameter
reverse-proxy_1  | time="2019-06-24T06:16:49Z" level=debug msg="Serving default certificate for request: \"test.localhost\""
reverse-proxy_1  | 2019/06/24 06:16:49 server.go:3012: http: TLS handshake error from 172.18.0.1:56356: remote error: tls: illegal parameter
```

---

[Someone has kindly spent the time looking into this error](https://hexeract.wordpress.com/2019/06/13/google-chrome-75-x-a-self-signed-certificate-and-err_ssl_key_usage_incompatible/) and discovered it's only affecting the recent v75 release of Chrome/Chromium, so users may only just start to experience this. It doesn't appear to be a bug in Chrome, but more strict handling of certs?

> As https is broken (at least for Google Chrome 75.x) I might as well create a new key/certificate pair without any keyUsage. Surely that can not be incompatible/incorrect

The code in question that seems to be related to this was committed [5 months ago](https://github.com/google/boringssl/commit/d7266ecc9bf92ffad277bc39653919da79c8f42b)(perhaps only just now making it to stable release):

> Generalizes ssl_cert_check_digital_signature_key_usage to check any part of the
key_usage, and adds a new error KEY_USAGE_BIT_INCORRECT for the generalized
method.

And [this is the bug report](https://bugs.chromium.org/p/chromium/issues/detail?id=795089) that the commit was fixing.

**Comment by polarathene** [comment]
There is also [this solution someone shared](https://superuser.com/questions/1451895/err-ssl-key-usage-incompatible-solution)(I haven't tested it personally):

> openssl req -new -x509 -days 36500 -nodes -newkey rsa:2048 -keyout cert.key -out cert.crt -extensions v3_req

**Comment by dtomcej** [comment]
Confirmed this today with the following config:

```toml
[global]
  checkNewVersion = false
  sendAnonymousUsage = false

[log]
  level = "DEBUG"

[entryPoints]
  [entryPoints.web-secure]
    address = ":4443"

[api]

[providers.file]
  filename = "dist/https_2.0.toml"

## dynamic configuration ##

[http.routers]
  [http.routers.router1]
    service = "service1"
    rule = "Host(`snitest.com`)"
    [http.routers.router1.tls]

[http.services]
  [http.services.service1]
    [http.services.service1.loadBalancer]
      [[http.services.service1.loadBalancer.servers]]
        url = "http://127.0.0.1:9010"
```

And running traefik via `./traefik --configfile=dist/https_2.0.toml`

I will be working on a PR to resolve tomorrow.

**Comment by dtomcej** [comment]
<img width="675" alt="Screen Shot 2019-07-22 at 7 54 19 PM" src="https://user-images.githubusercontent.com/6989327/61676807-86626280-acba-11e9-8e98-7d9e3fe926c1.png">

Sadface. Also no way to skip or bypass the error message.

**Comment by jcesclapez** [comment]
I updated from Alpha 8 to Beta 1 and same behavior, also i'm using acme letsencrypt no default cert.

**Comment by traefiker** [comment]
Closed by #5150.
